src/scheduler/pbs_sched_utils.cpp

```diff
@@ -684,13 +744,56 @@ sched_svr_init(void)
  * @return void
  */
 static void
-reconnect_servers()
+reconnect_servers(void)
 {
+	svr_conn_t **prim_conn = get_conn_svr_instances(clust_primary_sock);
+	svr_conn_t **sec_conn = get_conn_svr_instances(clust_secondary_sock);
+	int num_down = 0;
+	static time_t last_reconnect = -1;
+	const int min_reconnect_period = 30;	/* Reconnect at least 30secs after last */
+
 	pthread_mutex_lock(&cleanup_lock);
+	if (prim_conn == NULL || sec_conn == NULL || get_num_servers() == 1) {
+		close_servers();
+		connect_svrpool();
```

Comment:
Since connect_svrpool can also set part_tolerance to True, I feel we should always set it to False here if the number of servers configured is only 1.

Reply:
connect_svrpool() sets it to False the first thing, so we should be ok.